src/selection/highlight/visualSelection.js

```diff
@@ -136,6 +136,60 @@ class VisualSelection extends Selection {
     return startVisibleColumn;
   }
 
+  /**
+   * Searches the nearest visible column and row index, which is not hidden (is renderable). If one
+   * of the axes' range is entirely hidden, then created CellCoords object will hold the `null` value
+   * under a specific axis. For example, when we select the hidden column, then the calculated `col`
+   * prop will be `null`. In that case, rows are calculated further (regardless of the column result)
+   * to make rows header highlightable.
+   *
+   * @private
+   * @param {CellCoords} visualFromCoords Visual start coordinates for the range. Starting point for finding destination coordinates
+   * with visible coordinates (we are going from the starting coordinates to the end coordinates until the criteria are met).
+   * @param {CellCoords} visualToCoords Visual end coordinates for the range.
+   * @param {number} incrementByRow We are searching for a next visible rows by increasing (to be precise, or decreasing) indexes.
+   * This variable represent indexes shift. We are looking for an index:
+   * - for rows: from the left to the right (increasing indexes, then variable should have value 1) or
+   * other way around (decreasing indexes, then variable should have the value -1)
+   * - for columns: from the top to the bottom (increasing indexes, then variable should have value 1)
+   * or other way around (decreasing indexes, then variable should have the value -1).
+   * @param {number} incrementByColumn As above, just indexes shift for columns.
+   * @returns {CellCoords[]|null} Visual cell coordinates.
+   */
+  findVisibleCoordsInRangeSeparately(visualFromCoords, visualToCoords, incrementByRow, incrementByColumn) {
```

Comment:
What about:

```suggestion
  findVisibleHeaderRange(visualFromCoords, visualToCoords, incrementByRow, incrementByColumn) {
```

or

```suggestion
  getAdjustedHeaderRange(visualFromCoords, visualToCoords, incrementByRow, incrementByColumn) {
```

or

```suggestion
  findNullableEdgeRanges(visualFromCoords, visualToCoords, incrementByRow, incrementByColumn) {
```

Reply:
Thanks for your suggestions. I've stuck to this `findVisibleHeaderRange` https://github.com/handsontable/handsontable/pull/7053/commits/8a992fb3abf230658b7268ab22f5dfee6fcd3302.